### PR TITLE
Feature/fix usergroup update

### PR DESCRIPTION
**Pull request by idelcano**
…usergroup

User permission fixer was adding to the WIDP USERS group all the users without group. But was overriding the userGroup sharing settings, also was using the endpoint that broke our instances.

I have changed from post the full userGroup to addition in the usergroup the missing users

 I tested it in a local instance using:
 yarn  start usermonitoring run-permissions-fixer --config-file config.json

### CODE REVIEW

File: src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import { D2Api } from "@eyeseetea/d2-api/2.36";
+import { D2Api, Stats } from "@eyeseetea/d2-api/2.36";
 import log from "utils/log";
 import { PermissionFixerUserGroupExtended } from "domain/entities/user-monitoring/permission-fixer/PermissionFixerUserGroupExtended";
 import { PermissionFixerUserGroupRepository } from "domain/repositories/user-monitoring/permission-fixer/PermissionFixerUserGroupRepository";
 import { Async } from "domain/entities/Async";
 import { UserGroupNotFoundException } from "./exception/UserGroupNotFoundException";
+import { Id, Ref } from "domain/entities/Base";
```
Comment (nshandra) [inline src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:7]: `Id` is not used.

Comment (nshandra) [inline src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:7]: Id is not used.

File: src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts [hunk 2]
```diff
@@ -23,21 +24,45 @@ export class PermissionFixerUserGroupD2Repository implements PermissionFixerUser
             throw new UserGroupNotFoundException("Error getting user group: " + groupsIds);
         }
     }
-    async save(userGroup: PermissionFixerUserGroupExtended): Async<string> {
+    async save(userGroup: PermissionFixerUserGroupExtended, users: Ref[]): Async<string> {
         try {
-            const response = await this.api.models.userGroups.put(userGroup).getData();
-            if (response.status == "OK") {
+            const response = await this.appendUsersInUsergroup(userGroup, users);
```
Comment (nshandra) [inline src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:29]: Pedantic note: "in" usually means current location, "to" or "into" are more fitting.
So it should be `appendUsersToUsergroup` to accurately describe what the function does.

File: src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts [hunk 3]
```diff
@@ -23,21 +24,45 @@ export class PermissionFixerUserGroupD2Repository implements PermissionFixerUser
             throw new UserGroupNotFoundException("Error getting user group: " + groupsIds);
         }
     }
-    async save(userGroup: PermissionFixerUserGroupExtended): Async<string> {
+    async save(userGroup: PermissionFixerUserGroupExtended, users: Ref[]): Async<string> {
```
Comment (nshandra) [inline src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:27]: You can leave it as it is, but this is not saving a userGroup, only adding members to it.

File: src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts [hunk 4]
```diff
@@ -23,21 +24,45 @@ export class PermissionFixerUserGroupD2Repository implements PermissionFixerUser
...
+
+    private async appendUsersInUsergroup(
+        userGroup: PermissionFixerUserGroupExtended,
+        users: Ref[]
+    ): Async<string> {
+        const usersIds = users.map(({ id }) => ({ id: id }));
+        const response = await this.api
+            .post<UserGroupResponse>(
+                `/userGroups/${userGroup.id}/users/`,
+                { async: false },
```
Comment (nshandra) [inline src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:53]: Just a comment: as far as i know this does nothing as there is no async option for this endpoint.

**Review by nshandra — CHANGES_REQUESTED** [review]
Just a couple of small comments, otherwise looks good.

File: src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts (hunk 2 above)
Comment (idelcano) [reply src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:29]: na it's not  pedantic, thanks!

File: src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts (hunk 4 above)
Comment (idelcano) [reply src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:53]: the reason to add tye async false is due without it I got a typo error

Comment (nshandra) [reply src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts:53]: I see, probably was developed with `api/metadata` in mind

**Pull request merged**